Source/JavaScriptCore/wasm/WasmTable.cpp

```diff
@@ -196,17 +257,20 @@ FuncRefTable::FuncRefTable(uint32_t initial, std::optional<uint32_t> maximum)
         new (&m_importableFunctions.get()[i]) Function();
         ASSERT(m_importableFunctions.get()[i].m_function.typeIndex == Wasm::TypeDefinition::invalidIndex); // We rely on this in compiled code.
         ASSERT(!m_importableFunctions.get()[i].m_instance);
+        ASSERT(m_importableFunctions.get()[i].m_value.isNull());
     }
 }
 
-void FuncRefTable::setFunction(uint32_t index, JSObject* optionalWrapper, Function function)
+void FuncRefTable::setFunction(uint32_t index, JSObject* optionalWrapper, WasmToWasmImportableFunction function, Instance* instance)
 {
     RELEASE_ASSERT(index < length());
     RELEASE_ASSERT(m_owner);
-    clear(index);
+    m_importableFunctions.get()[index].m_function = function;
+    m_importableFunctions.get()[index].m_instance = instance;
     if (optionalWrapper)
-        m_jsValues.get()[index].set(m_owner->vm(), m_owner, optionalWrapper);
-    m_importableFunctions.get()[index] = function;
+        m_importableFunctions.get()[index].m_value.set(m_owner->vm(), m_owner, optionalWrapper);
+    else
+        m_importableFunctions.get()[index].m_value.set(m_owner->vm(), m_owner, jsNull());
```

Comment:
Is it worth pre-caching `m_importableFunctions.get()[index]` in a local?

Secondly, why not use a `?:` expression (instead of the explicit `if` statement) for setting `m_value`?

Reply:
Also, I think probably the second one can use setEarlyValue. Changed.